devtools/make_release

```diff
@@ -0,0 +1,350 @@
+UPDATE_VERSION_CMD = "perl -pi -e \"s/{version_str}{old_version}/" \
+                     "{version_str}'{new_version}'/\" {file}"
+
+COMMIT_MESSAGE = 'Release {version}'
+
+PY_BUILD_CMD = 'python setup.py bdist_wheel --universal --dist-dir {}'
+PY_VERSION_STR = "__version__ = "
+
+JS_BUILD_CMD = 'yarn {target}'
+JS_UPDATE_VERSION_CMD = 'npm version "{version}" --no-git-tag-version'
+
+GO_BUILD_CMD = 'govendor sync; govendor update +vendor; ' \
+               'go build -o {out} {target}'
+
+# Files updated to consider for commit or rollback
+UPDATE_FILES = []
+
+
+def do_command(cmd):
+    """Execute a shell command when can't use a python library."""
+    return sp.call(cmd, shell=True)
+
+
+def do_package_command(package, cmd):
+    """Execute a shell command for a package."""
+    log.debug(' will execute command : {}'.format(cmd))
+    errno = sp.call(cmd, shell=True)
+    if errno == 0:
+        log.info('Build package {} OK'.format(package['name']))
+        return True
+    msg = 'Build package {} failed: {}'.format(package['name'], errno)
+    raise Exception(msg)
+
+
+def check_versions(current, new):
+    """Compare current and new version, raise if new is not superior."""
+    t_new = tuple(int(x) for x in new.split('.'))
+    t_cur = tuple(int(x) for x in current.split('.'))
+    if t_new <= t_cur:
+        raise Exception('Current version {} and new expected one {} mismatch'.
+                        format(current, new))
+
+
+def update_javascript_package(package, package_dir, version):
+    """Update javascript version in package.json."""
+    pkg_file = '{}/package.json'.format(package_dir)
+    if not os.path.isfile(pkg_file):
+        raise Exception('File {} for package version not found'.
+                        format(pkg_file))
+    current = None
+    with open(pkg_file) as v_file:
+        current = json.loads(v_file.read()).get('version')
+    if not current:
+        raise Exception('No current version found for package {}'.
+                        format(package['name']))
+    log.info('Current version found for package {} is {}'.
+             format(package['name'], current))
+    check_versions(current, version)
+    version_cmd = JS_UPDATE_VERSION_CMD.format(version=version)
+    cmd = 'cd {};{}'.format(package_dir, version_cmd)
+    res = do_package_command(package, cmd)
+    if res:
+        UPDATE_FILES.append(pkg_file)
+    return res
+
+
+def make_javascript_package(package, directory, version, output_directory):
+    """Build javascript distribution."""
+    if not os.path.isdir(directory):
+        raise Exception('No such package directory {}'.format(directory))
+
+    # Should fail if using package name
+    build_cmd = JS_BUILD_CMD.format(target=package['build_target'])
+    cmd = 'cd {}; {}'.format(directory, build_cmd)
+    res = do_package_command(package, cmd)
+    if not res:
+        return False
+    # Move build distribution
```

Comment:
I introduce a build_target key for javascript and golang package configuration so we can have different packages build from the same directory (different package name and build_target), but we must have a predicatable way where the build can be found.

I use package['name'] but I'm not sure it will always the case ( web_application/dist/<package_name>)
May add another configuration key (or get it from the package.json ?)